### ISSUE TITLE
Unable to restore Shield from backups

### ISSUE THREAD

**Issue by brunograz** [shieldproject/shield#492]
Shield version: 8.0.15

Using Shield to backup itself with the fs plugin on /var/vcap/store/shield - two separate jobs, both with and without compression. 

After downloading the archive from the store a new installation of Shield was created, but when I tried to restore using the archive plus the fixed key the initialization failed and the only error log in Shield is shown below. 

Error log:
"""
2019/02/11 20:31:53 [error] 4449#0: *217 client intended to send too large body: 1109041 bytes, client: <redacted>, server: _, request: "POST /v2/bootstrap/restore HTTP/1.1", host: "<redacted>", referrer: "<redacted>"
"""

**Comment by jhunt** [comment]
How big is the backup archive?  It sounds like nginx is denying the request as too large.

**Comment by brunograz** [comment]
Hi @jhunt 

The backup archive generated of /var/vcap/store/shield is ~11MB compressed and roughly 14MB decompressed.

**Comment by jhunt** [comment]
Looks like we set client_max_body_size to 10m, here: https://github.com/starkandwayne/shield-boshrelease/blob/master/jobs/core/templates/config/nginx.conf#L93

Can you get on the SHIELD VM, manually change that to something higher, like 20m, `monit restart nginx` and try again?

**Comment by brunograz** [comment]
I will try later today and let you know.

Thanks for the prompt reply!

**Comment by brunograz** [comment]
Hi @jhunt 

I have increased the client_max_body_size to 30m in /var/vcap/jobs/core/config/nginx.conf and restart nginx (as well as the other processes) but the issue is still happening - same as described above. 

The UI keeps looping at "SHIELD initialization failed, restarting initialization process..."

**Comment by jhunt** [comment]
So the interesting bit here is this:
```
client intended to send too large body: 1109041 bytes
```

1,109,041 bytes is 1.1M, not 11M.  I wonder where that number is coming from...

I'll try to get some lab time to spin this up and do stuff.

**Comment by brunograz** [comment]
Managed to upload the backup by adding the client_max_body_size directly underneath {server} in nginx.conf. 
This line exactly: https://github.com/starkandwayne/shield-boshrelease/blob/master/jobs/core/templates/config/nginx.conf#L64.

Although the restore procedure started, it failed at the same step as mentioned in https://github.com/starkandwayne/shield/issues/443.

```
Running restore task (using bzip2 compression)
==============================================
panic: cipher.NewCTR: IV length must equal block size

goroutine 1 [running]:
crypto/cipher.NewCTR(0x7df440, 0xc420064de0, 0xc4200ec000, 0xf0, 0xf0, 0x0, 0x0)
	/goroot/src/crypto/cipher/ctr.go:38 +0x322
github.com/starkandwayne/shield/crypter.Stream(0xc4200144d0, 0xa, 0xc4200144e0, 0x10, 0x10, 0xc4200ec000, 0xf0, 0xf0, 0x1e0, 0xc42004df50, ...)
	/tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/crypter/crypter.go:47 +0x426
main.main()
	/tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/cmd/shield-crypt/main.go:40 +0x4fb

bunzip2: Compressed file ends unexpectedly;
	perhaps it is corrupted?  *Possible* reason follows.
bunzip2: Inappropriate ioctl for device
	Input file = (stdin), output file = (stdout)

It is possible that the compressed file(s) have become corrupted.
You can use the -tvv option to test integrity of such files.

You can use the `bzip2recover' program to attempt to recover
data from undamaged sections of corrupted files.
```

Any solution for this issue?

**Comment by jhunt** [comment]
That looks like a problem with the encryption parameters.

**Comment by jhunt** [comment]
In the process of rewriting the task scheduler (which the self-restore was an expertly ugly set of edge cases), I believe we found some mismatch between the handling of the fixed key material and the derivation of key+iv for encryption/decryption.

This has been fixed and vigorously tested in v8.1., due out next week at CF Summit Philadelphia.  Existing Fixed Keys and their associated archives should be viable after the fixes in 8.1. are deployed.

**Comment by jhunt** [comment]
@patjones and I worked through testing this, rigorously, in our lab, and noticed a few issues related to pathing that have been fixed in #525.  We will be documenting this process, and the internal details of how it works, in a forthcoming blog post, in advance of an 8.2 release.  I'll drop a note here when that release happens, so you can try that, @brunograz

**Comment by jhunt** [comment]
Whelp, this one fell off my radar.  @brunograz have you per chance seen and/or tried to deploy the 8.2.1 SHIELD to see if this fixes your issues?
